File: Sorting/selection_sort/selection_sort.py
def selection_sort(arr):

    for i in range(len(arr)-1):
        mini_index = i
        for j in range(i, len(arr)):
            if arr[j] < arr[mini_index]:
                mini_index = j

        arr[i], arr[mini_index] = arr[mini_index], arr[i]

    print(arr)

arr = [13, 46, 24, 52, 20, 9]

# Call selection sort
selection_sort(arr)

File: Sorting/selection_sort/selection_sort.cpp
#include<bits/stdc++.h>
using namespace std;
void selectionSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++) {
        int minIndex = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }
        int temp = arr[minIndex];
        arr[minIndex] = arr[i];
        arr[i] = temp;
    }

    cout << "After selection sort: " << "\n";
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << "\n";
}

int main() {
    // Initialize array
    int arr[] = {13, 46, 24, 52, 20, 9};
    int n = sizeof(arr) / sizeof(arr[0]);

    // Print array before sorting
    cout << "Before selection sort: " << "\n";
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << "\n";

    // Call selection sort
    selectionSort(arr, n);

    return 0;
}



